application/src/main/java/org/thingsboard/server/exception/ThingsboardErrorResponseHandler.java

```diff
@@ -71,6 +74,17 @@ public void handle(Exception exception, HttpServletResponse response) {
             try {
                 response.setContentType(MediaType.APPLICATION_JSON_VALUE);
 
+                String cause = "";
+                if (exception.getCause() != null) {
+                    cause = exception.getCause().getClass().getCanonicalName();
+                }
+                if (exception instanceof IllegalArgumentException || exception instanceof IncorrectParameterException
+                        || exception instanceof DataValidationException || cause.contains("IncorrectParameterException")) {
+                    exception = new ThingsboardException(exception.getMessage(), ThingsboardErrorCode.BAD_REQUEST_PARAMS);
+                } else if (exception instanceof MessagingException) {
+                    exception = new ThingsboardException("Unable to send mail: " + exception.getMessage(), ThingsboardErrorCode.GENERAL);
+                }
```

Comment:
Exceptions from controllers will not be logged according to BaseController.handleThingsboardException logic
```
if (logException && logControllerErrorStackTrace) {
    log.error("Error [{}]", exception.getMessage(), exception);
}
```